pynumdiff/optimize/_optimize.py

```diff
@@ -0,0 +1,186 @@
+"""Optimization"""
+import scipy.optimize
+import numpy as np
+from itertools import product
+from functools import partial
+from warnings import filterwarnings
+from multiprocessing import Pool
+
+from ..utils import evaluate
+from ..finite_difference import first_order
+from ..smooth_finite_difference import mediandiff, meandiff, gaussiandiff, friedrichsdiff, butterdiff, splinediff
+from ..linear_model import spectraldiff, polydiff, savgoldiff, lineardiff
+from ..total_variation_regularization import velocity, acceleration, jerk, iterative_velocity, smooth_acceleration, jerk_sliding
+from ..kalman_smooth import constant_velocity, constant_acceleration, constant_jerk
+
+
+# Map from method -> (init_conds, bounds_low_hi)
```

Comment:
All the wrapper classes basically just encoded the information in this big dictionary, but much more spread out. Here it's all in one place.